File: algorithms/leetcode/268-missing-number/MissingNumber.java
public class MissingNumber {
    public int missingNumber(int[] nums) {
        int result = 0;

        int n = nums.length;

        for (int i = 0; i < n; i++) {
            result ^= i ^ nums[i];
        }

        result ^= n;

        return result;
    }
}

File: leetcode/268-missing-number/missing-number.cpp
class Solution {
public:
  int missingNumber(vector<int>& nums) {
    int i = 0;
    while (i < nums.size()) {
      if (nums[i] < nums.size() && nums[i] != nums[nums[i]]) {
        swap(nums[i], nums[nums[i]]);
      } else {
        i++;
      }
    }

    for (int i = 0; i < nums.size(); i++) {
      if (nums[i] != i) {
        return i;
      }
    }

    return nums.size();
  }
};
